plugins/azure/resoto_plugin_azure/azure_client.py

```
@@ -133,30 +133,40 @@ def _update_or_delete_tag(self, tag_name: str, tag_value: str, resource_id: str,
 
     # noinspection PyProtectedMember
     def _call(self, spec: AzureApiSpec, **kwargs: Any) -> List[Json]:
-        _SERIALIZER = Serializer()
+        ser = Serializer()
 
         error_map = {
             401: ClientAuthenticationError,
             404: ResourceNotFoundError,
         }
 
+        # Construct lookup map used to fill query and path parameters
+        lookup_map = {"subscriptionId": self.subscription_id, "location": self.location, **kwargs}
+
         # Construct headers
-        headers = case_insensitive_dict(kwargs.pop("headers", {}) or {})
-        headers["Accept"] = _SERIALIZER.header("accept", headers.pop("Accept", "application/json"), "str")  # type: ignore # noqa: E501
+        headers = case_insensitive_dict()
+        headers["Accept"] = ser.header("accept", headers.pop("Accept", "application/json"), "str")  # type: ignore # noqa: E501
+
+        # Construct path map
+        path_map = case_insensitive_dict()
+        for param in spec.path_parameters:
+            if lookup_map.get(param, None) is not None:
```

if there is no value in the lookup_map, we should throw an error